Console/Command/ResetYotpoSync.php

```diff
@@ -0,0 +1,225 @@
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Magento\Framework\App\State as AppState;
+use Yotpo\Core\Model\Config;
+use Yotpo\Core\Model\Sync\Reset;
+
+/**
+ * Class ReSetYotpoSync - Manage Yotpo Reset Sync
+ */
+class ResetYotpoSync extends Command
+{
+    const YOTPO_ENTITY              = 'entity';
+    const STORE_ID                  = 'store_id';
+    const YOTPO_ENTITY_ALL          = 'all';
+    const YOTPO_ENTITY_CUSTOMERS    = 'customer';
+    const YOTPO_ENTITY_ORDERS       = 'order';
+    const YOTPO_ENTITY_CATALOG      = 'catalog';
+
+    /**
+     * @var string[]
+     */
+    protected $yotpoEntities = ['order', 'customer', 'catalog', 'all'];
+
+    /**
+     * @var AppState
+     */
+    protected $appState;
+
+    /**
+     * @var Reset
+     */
+    protected $syncReset;
+
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * RetryYotpoSync constructor.
+     * @param AppState $appState
+     * @param Reset $syncReset
+     * @param string|null $name
+     */
+    public function __construct(
+        Reset $syncReset,
+        AppState $appState,
+        Config $config,
+        string $name = null
+    ) {
+        $this->syncReset = $syncReset;
+        $this->appState = $appState;
+        $this->config = $config;
+        parent::__construct($name);
+    }
+
+    /**
+     * Configures the current command.
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $options = [
+            new InputOption(
+                self::YOTPO_ENTITY,
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Entity'
+            ),
+            new InputOption(
+                self::YOTPO_ENTITY_ALL,
+                null,
+                InputOption::VALUE_NONE,
+                'All Entities'
+            ),
+            new InputOption(
+                self::STORE_ID,
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Store ID'
+            ),
+        ];
+
+        $this->setName('yotpo:resetsync')
+            ->setDescription('Reset Sync')
+            ->setDefinition($options);
+
+        parent::configure();
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return $this|int
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->setAreaCode();
+
+        $isAllOption = $input->getOption(self::YOTPO_ENTITY_ALL);
+        $storeId = $input->getOption(self::STORE_ID);
+        if ($storeId) {
+            $storeIds = [$storeId];
+        } else {
+            $storeIds = $this->config->getAllStoreIds();
+        }
+        if (!$storeIds) {
+            return $this;
+        }
+        foreach ($storeIds as $storeId) {
+            if ($isAllOption) {
```

Comment:
It should be passed like below
bin/magento yotpo:resetsync --entity=all

It will then reset the sync for all the entities.

If you need to reset the sync for a particular entity , it should be passed like below.
bin/magento yotpo:resetsync --entity=order
bin/magento yotpo:resetsync --entity=customer
bin/magento yotpo:resetsync --entity=catalog

Reply:
I'm sorry, but this doesn't seem completely right according to the code.

`YOTPO_ENTITY_ALL` is an input option on its own, the fact that you're checking it over here, means that there's a duplication of logic.
Resetting all attributes can be done via the `--all` `InputOption` you've created, and also from the `--entity` by passing `all` as value.
Let's stick with one of them, which is passing `all` value to `--entity`.

Reply:
I'm sorry, but this doesn't seem completely right according to the code.

`YOTPO_ENTITY_ALL` is an input option on its own, the fact that you're checking it over here, means that there's a duplication of logic.
Resetting all attributes can be done via the `--all` `InputOption` you've created, and also from the `--entity` by passing `all` as value.
Let's stick with one of them, which is passing `all` value to `--entity`.

Reply:
Okay, we will make these changes

Reply:
Okay, we will make these changes